### PR TITLE
Fixed default help being displayed when a custom help exists

### PR DESCRIPTION
**Pull request by RagingLink**
Changed the order of checking the argument from default command then custom command, to custom command then default command.
Fixes bug report [728](https://airtable.com/shrEUdEv4NM04Wi7O/tblyFuWE6fEAbaOfo/viwDg5WovcwMA9NIL/recaBJMuEGz0MOqoY?blocks=bipip3uBZFRPcSy3C)

**Changed**:
- Order of if else [ 16b3a5d ](https://github.com/blargbot/blargbot/commit/16b3a5df1f0a52bd3aeb5d219d59f607a3cc7825)

### CODE REVIEW

File: src/dcommands/help.js [hunk 1]
```diff
@@ -31,31 +31,34 @@ class HelpCommand extends BaseCommand {
...
-                if (instance.flags && instance.flags.length > 0)
-                    embed.fields.push({
-                        name: '**Flags**',
-                        value: instance.flags.map(flag => `\`-${flag.flag}\` or \`--${flag.word}\` - ${flag.desc}`).join('\n')
-                    });
-            } else {
-                let helpText = await bu.ccommand.gethelp(msg.guild.id, words[1]);
-                if (helpText) {
+            let customCommand = await bu.ccommand.get(msg.guild.id, words[1]);
+            if(customCommand) {
```
Comment (HunteRoi) [inline src/dcommands/help.js:35]: I'd add a simple space between the if and the first (

File: src/dcommands/help.js [hunk 2]
```diff
@@ -31,31 +31,34 @@ class HelpCommand extends BaseCommand {
...
+                let helpText = customCommand.help;
+                if(helpText) {
```
Comment (HunteRoi) [inline src/dcommands/help.js:37]: here too

File: src/dcommands/help.js [hunk 3]
```diff
@@ -31,31 +31,34 @@ class HelpCommand extends BaseCommand {
...
-                    embed.fields.push({
-                        name: '**Flags**',
-                        value: instance.flags.map(flag => `\`-${flag.flag}\` or \`--${flag.word}\` - ${flag.desc}`).join('\n')
-                    });
-            } else {
-                let helpText = await bu.ccommand.gethelp(msg.guild.id, words[1]);
-                if (helpText) {
+            let customCommand = await bu.ccommand.get(msg.guild.id, words[1]);
+            if(customCommand) {
+                let helpText = customCommand.help;
```
Comment (HunteRoi) [inline src/dcommands/help.js:36]: Can be a `const` as the `helpText` variable is not overwritten somewhere else.

File: src/dcommands/help.js [hunk 4]
```diff
@@ -31,31 +31,34 @@ class HelpCommand extends BaseCommand {
...
-                    });
-                if (instance.flags && instance.flags.length > 0)
-                    embed.fields.push({
-                        name: '**Flags**',
-                        value: instance.flags.map(flag => `\`-${flag.flag}\` or \`--${flag.word}\` - ${flag.desc}`).join('\n')
-                    });
-            } else {
-                let helpText = await bu.ccommand.gethelp(msg.guild.id, words[1]);
-                if (helpText) {
+            let customCommand = await bu.ccommand.get(msg.guild.id, words[1]);
```
Comment (HunteRoi) [inline src/dcommands/help.js:34]: Can be a `const` as the `ccommand` variable is not overwritten somewhere else.

File: src/dcommands/help.js [hunk 5]
```diff
@@ -31,31 +31,34 @@ class HelpCommand extends BaseCommand {
...
+                let helpText = customCommand.help;
+                if(helpText) {
                     embed.color = this.getColor('CUSTOM');
                     embed.title = `Help for ${words[1].toLowerCase()} (Custom Command)`;
                     embed.description = helpText;
                 }
+            } else {
+                if (CommandManager.commandList.hasOwnProperty(words[1]) && !CommandManager.commandList[words[1]].hidden
+                && (!CommandManager.commandList[words[1]].onlyOn || CommandManager.commandList[words[1]].onlyOn === msg.guild.id)) {
+                    let instance = CommandManager.built[CommandManager.commandList[words[1]].name];
```
Comment (HunteRoi) [inline src/dcommands/help.js:45]: Can be a `const` as the `instance` variable is not overwritten somewhere else.

File: src/dcommands/help.js [hunk 6]
```diff
@@ -31,31 +31,34 @@ class HelpCommand extends BaseCommand {
...
+                    let definition = CommandManager.commandList[words[1]];
```
Comment (HunteRoi) [inline src/dcommands/help.js:46]: Can be a `const` as the `definition` variable is not overwritten somewhere else.

**Review by HunteRoi — CHANGES_REQUESTED** [review]

File: src/dcommands/help.js (hunk 5 above)
Comment (RagingLink) [reply src/dcommands/help.js:45]: I left this bit of code untouched as I believe the code should only be changed if necessary, but I could change it I guess 🤷 Just like I don't/can't really use a formatter when making these pull requests as it changes pretty much every single line

File: src/dcommands/help.js (hunk 6 above)
Comment (RagingLink) [reply src/dcommands/help.js:46]: Same for my reply above

**Review by HunteRoi — APPROVED** [review]

**Pull request merged**